bundles/org.openhab.core.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/config/EnrichedConfigDescriptionDTO.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.rest.core.config;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.core.ConfigDescription;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterGroupDTO;
+
+/**
+ * This is an enriched data transfer object that is used to serialize {@link ConfigDescription}s.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class EnrichedConfigDescriptionDTO extends ConfigDescriptionDTO {
```

Comment:
Hm, I must be missing something. What is the difference between the `EnrichedConfigDescriptionDTO`and the normal `ConfigDescriptionDTO `? Why do we need this additional class at all?

Reply:
I introduced this class to have a clear hierarchy of DTOs and to use it for the response definition of the REST API endpoints.

But in general you are right. The important thing is the extended `EnrichedConfigDescriptionParameterDTO`.

Reply:
Then I'd prefer to remove this class.

Reply:
Done.